File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/ArtistCredit.java
package org.metabrainz.mobile.data.sources.api.entities;

import org.metabrainz.mobile.data.sources.api.entities.mbentity.Artist;

public class ArtistCredit {
    private String name;
    private String joinphrase;
    private Artist artist;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getJoinphrase() {
        return joinphrase;
    }

    public void setJoinphrase(String joinphrase) {
        this.joinphrase = joinphrase;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }
}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/ArtistCredit.kt
package org.metabrainz.mobile.data.sources.api.entities

import org.metabrainz.mobile.data.sources.api.entities.mbentity.Artist

class ArtistCredit {
    var name: String? = null
    var joinphrase: String? = null
    var artist: Artist? = null
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ArtistCredit) return false
        return name == other.name && joinphrase == other.joinphrase && artist == other.artist
    }

    override fun hashCode(): Int {
        var result = name?.hashCode() ?: 0
        result = 31 * result + (joinphrase?.hashCode() ?: 0)
        result = 31 * result + (artist?.hashCode() ?: 0)
        return result
    }
}